### ISSUE TITLE
`TxId` means two different things.

### ISSUE THREAD

**Issue by zmrocze** [input-output-hk/plutus-apps#443]
### Summary

`TxId` from `plutus-ledger-api-0.1.0.0` can be at least two different things, according to usages in `plutus-apps`:
 - double-sha256 hash of a plutus transaction stripped from input witnesses. As documented.
 - but also a cardano TxId, the bytestring refering to tx in raw blockchain data. This is different as first of all it is a blake2b_256 hash.
 
One can get the first using [txId](https://github.com/input-output-hk/plutus-apps/blob/59bf8bd788953760e21d3a9f2b4f56f0371a653f/plutus-ledger/src/Ledger/Tx.hs#L196) exported from plutus-ledger.

The second is for example what is visible inside `TxInfo` context of a plutus validation script. 

Then there is [getCardanoTxId](https://github.com/input-output-hk/plutus-apps/blob/59bf8bd788953760e21d3a9f2b4f56f0371a653f/plutus-ledger/src/Ledger/Tx.hs#L160) or getCardanoApiTxId which act on `These Plutus.V1.Ledger.Tx.Tx Cardano.Api.Tx` (more or less) by applying one or the other hashing function (double-sha256 of the Plutus.Tx for Both). It is used for example in [Tutorial.Escrow](https://github.com/input-output-hk/plutus-apps/blob/03ba6b7e8b9371adf352ffd53df8170633b6dffa/plutus-use-cases/src/Plutus/Contracts/Tutorial/Escrow.hs#L243).

### Steps to reproduce the behavior

_No response._

### Actual Result

There are no clear invariants of `TxId`.

### Expected Result

Is this expected? I can't asses if this is problematic, but certainly looks so.

### System info

Plutus-0.1.0.0
Plutus-apps: fdb0eec47

**Comment by koslambrou** [comment]
Good point. We only use `txId` in `plutus-ledger` for the emulator (testing plutus contracts). This function shouldn't be used with transaction that are going to be submitted in the real blockchain.

Eventually, `Plutus.Tx` and `txId` will be deleted and replace with `Cardano.Api.Tx` types.
